src/certs.rs

```diff
@@ -0,0 +1,258 @@
+    };
+    let initial_config =
+        build_tls_server_config(cert.clone(), key.clone_key(), client_verifier.clone())?;
+
+    let rust_config = RustlsConfig::from_config(Arc::new(initial_config));
+    let reloadable_rust_config = rust_config.clone();
+
+    // Init inotify to watch for changes in the certificate files
+    let inotify =
+        inotify::Inotify::init().map_err(|e| anyhow!("Cannot initialize inotify: {e}"))?;
+    let cert_watch = inotify
+        .watches()
+        .add(
+            tls_config.cert_file.clone(),
+            inotify::WatchMask::CLOSE_WRITE,
+        )
+        .map_err(|e| anyhow!("Cannot watch certificate file: {e}"))?;
+    let key_watch = inotify
+        .watches()
+        .add(tls_config.key_file.clone(), inotify::WatchMask::CLOSE_WRITE)
+        .map_err(|e| anyhow!("Cannot watch key file: {e}"))?;
+
+    let client_ca_watches: Result<Vec<WatchDescriptor>, anyhow::Error> = tls_config
+        .client_ca_file
+        .clone()
+        .into_iter()
+        .map(|path| {
+            inotify
+                .watches()
+                .add(path, inotify::WatchMask::CLOSE_WRITE)
+                .map_err(|e| anyhow!("Cannot watch client certificate file: {e}"))
+        })
+        .collect();
+
+    let client_ca_watches = client_ca_watches?;
+
+    let buffer = [0; 1024];
+    let stream = inotify
+        .into_event_stream(buffer)
+        .map_err(|e| anyhow!("Cannot create inotify event stream: {e}"))?;
+
+    tokio::spawn(async move {
+        tokio::pin!(stream);
+        let mut cert_changed = false;
+        let mut key_changed = false;
+        let mut client_ca_changed = false;
+
+        while let Some(event) = stream.next().await {
+            let event = match event {
+                Ok(event) => event,
+                Err(e) => {
+                    warn!("Cannot read inotify event: {e}");
+                    continue;
+                }
+            };
+
+            if event.wd == cert_watch {
+                info!("TLS certificate file has been modified");
+                cert_changed = true;
+            }
+            if event.wd == key_watch {
+                info!("TLS key file has been modified");
+                key_changed = true;
+            }
+
+            for client_ca_watch in client_ca_watches.iter() {
+                if event.wd == *client_ca_watch {
+                    info!("TLS client certificate file has been modified");
+                    client_ca_changed = true;
+                }
+            }
+
+            // Reload the client CA certificates if they have changed, keeping the current server certificates unchanged
+            if client_ca_changed {
+                info!("Reloading Client CA certificates");
+
+                client_ca_changed = false;
+
+                match load_client_ca_certs(tls_config.client_ca_file.clone()).await {
+                    Ok(cv) => {
+                        client_verifier = Some(cv);
```

Comment:
I think you are using a variable from outside the scope of the task because the variable is moved to the task scope. Right? Have you did that intentionally, right? The same apply for the `cert` and `key` variable some lines below. Just double checking. No problem for me so far.

Reply:
The variable is the moved outer scope `mut` variable. The outer scope performs the first initialization. 
This is needed to preserve the state of the certificates/client CA so we can perform a partial reload.

Reply:
Yes, I'm fine with that. As I said, I was just double checking if that was on purpose.